src/gitMenuCommands.ts

```diff
@@ -68,7 +89,11 @@ export function addCommands(app: JupyterLab, services: ServiceManager) {
           terminal.session.send({
             type: 'stdin',
             content: [
-              'cd "' + currentFileBrowserPath.split('"').join('\\"') + '"\n'
+              'cd "' +
+                app.info.directories.serverRoot +
```

Comment:
I'm not 100% sure if this is the best way to get the Jupyter server root. Open to suggestions.

Reply:
That serverRoot variable is for UI information only, and should not be relied on in code (the server may choose to make it an empty string, for example, or the root "file system" may be provided by a database, not corresponding to a directory on the file system). See https://github.com/jupyterlab/jupyterlab/blob/cf1063590f3f438b771b679fafc2eeed0838421e/packages/application/src/frontend.ts#L278-L303 for more complete documentation.

In general, working with directories on the server outside of the sandbox of the server root should be done by a server extension.

Reply:
Thanks @jasongrout for that clarification. @jaipreet-s So we could make an endpoint in the server extension that returns the absolute path to the git directory.

Reply:
Thanks. This is on my TODO list to get back to.

Reply:
> Thanks @jasongrout for that clarification. @jaipreet-s So we could make an endpoint in the server extension that returns the absolute path to the git directory.

@saulshanabrook In #395, I've added a server endpoint to return the server root which can be used for this as well.